packages/framework/src/Component/Elasticsearch/IndexFacade.php

```diff
@@ -237,18 +205,9 @@ public function migrate(IndexDefinition $indexDefinition, OutputInterface $outpu
 
         $output->writeln(sprintf('Migrating index "%s" on domain "%s"', $indexName, $domainId));
 
-        try {
-            $this->indexRepository->createIndex($indexDefinition);
-        } catch (ElasticsearchIndexAlreadyExists $exception) {
-            $output->writeln(sprintf(
-                'Index "%s" was not created on domain "%s" because it has already exist',
-                $indexDefinition->getIndexName(),
-                $indexDefinition->getDomainId()
-            ));
-        }
-
+        $this->createIndexWhenNeeded($indexDefinition, $output);
         $this->indexRepository->reindex($existingIndexName, $indexDefinition->getVersionedIndexName());
```

Comment:
You have created here several encapsulating methods for repository methods adding info message - it would be nice add similar ones for reindex and index delete. What do you think?

Reply:
I really have no clue what you are suggesting. Can you be more specific?

Reply:
Ah, ok... I do not se any reason to encapsulate `reindex` and `deleteIndex` as it will not bring any advantage.